### PR TITLE
pd: add timestamp field for region heartbeat

### PR DESCRIPTION
**Pull request by Connor1996**
record time for pd to determine whether heartbeat is stall or not

### CODE REVIEW

File: src/pd/client.rs [hunk 1]
```diff
@@ -301,6 +301,10 @@ impl PdClient for RpcClient {
         req.set_keys_read(region_stat.read_bytes);
         req.set_approximate_size(region_stat.approximate_size);
 
+        let now = SystemTime::now();
```
Comment (siddontang) [inline src/pd/client.rs]: do we have a coarse system time? @overvenus

Comment (overvenus) [reply src/pd/client.rs]: Nope, the coarse version should be `clock_gettime(CLOCK_REALTIME_COARSE)`.

**Comment by siddontang** [comment]
LGTM
PTAL @disksing @overvenus

File: src/pd/client.rs [hunk 2]
```diff
@@ -301,6 +301,10 @@ impl PdClient for RpcClient {
...
+        let ts = now.duration_since(UNIX_EPOCH).expect("time went backwards").as_secs();
```
Comment (disksing) [inline src/pd/client.rs]: Milliseconds should be better since metrics bucket on pd side starts from 1ms.

File: src/pd/client.rs [hunk 3]
```diff
@@ -301,6 +301,9 @@ impl PdClient for RpcClient {
         req.set_keys_read(region_stat.read_bytes);
         req.set_approximate_size(region_stat.approximate_size);
 
+        let now = realtime_coarse_now();
+        req.set_timestamp((now.sec * 1_000) as u64 + (now.nsec / 1_000_000) as u64);
```
Comment (siddontang) [inline src/pd/client.rs]: I remember we have a time to millisecond function.

File: src/util/time.rs [hunk 4]
```diff
@@ -182,6 +184,17 @@ mod inner {
         // TODO Add monotonic coarse clock time impl for macos and windows
         monotonic_raw_now()
     }
+
+    pub fn realtime_now() -> Timespec {
```
Comment (siddontang) [inline src/util/time.rs]: please add test

Comment (siddontang) [reply src/util/time.rs]: I suggest still using SystemTime directly, you can add this later if possible.

Comment (Connor1996) [reply src/util/time.rs]: okay.

**Pull request merged**